app/controllers/images_controller.rb

```diff
@@ -20,6 +21,14 @@ def destroy
   def edit
     @image = Image.find(params[:id])
     @gallery = Gallery.find(params[:gallery_id])
+    @tags = Tag.all
+  end
+  
+  def update
+    @gallery = Gallery.find(params[:gallery_id])
+    @image = Image.find(params[:id])
+    @image.update_attributes(image_params)
+    redirect_to [@gallery, @image]
```

Comment:
If you are redirecting then there is no need to make these instance variables. When you redirect, the user comes back and a whole new action happens, so no view is rendered for this. We only need instance variables when we are rendering a view. So this would become:

``` ruby
gallery = Gallery.find(params[:gallery_id])
image = Image.find(params[:id])
image.update_attributes(image_params)
redirect_to [gallery, image]
```